components/CodeSnippet.vue

```diff
@@ -64,20 +75,109 @@ const copied = ref(false);
 
 defineExpose({ modalVisible });
 
-const parameters = computed(() => appStore.currentScenario.parameters);
+const everyScenarioHasParameters = computed(() => props.scenarios.every(scenario => !!scenario.parameters));
+
+// TODO: (jidea-317) In the future, we should have the R API capture the actual call made to the model,
```

Comment:
since this ticket will be done, makes even more sense to extrac the logic out into another file so can be easily replaced

Reply:
Nice one, have extracted into a composable `useCodeSnippet`